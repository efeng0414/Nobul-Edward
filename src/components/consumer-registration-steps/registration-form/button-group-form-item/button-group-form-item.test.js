import React from "react";
import ButtonGroupFormItem from "./index";

describe("ButtonGroupFormItem Component", () => {
  const componentProps = {
    formHasSubmitted: true,
    intl: global.getIntl(),
    signUpUser: () => {},
    toPreviousPage: () => {}
  };

  const componentJSX = <ButtonGroupFormItem {...componentProps} />;

  describe("Component Snapshot", () => {
    const componentSnapshot = global.renderer.create(componentJSX).toJSON();
    it("should match stored snapshot", () => {
      expect(componentSnapshot).toMatchSnapshot();
    });
  });

  describe("Component Render", () => {
    const componentMount = global.Enzyme.shallow(componentJSX);

    it("should call signUpUserProp on button next click", () => {
      componentProps.signUpUser = jest.fn();
      componentMount.setProps(componentProps);
      componentMount.find("Button.button-next").simulate("click");

      expect(componentProps.signUpUser).toHaveBeenCalled();
    });
  });
});
