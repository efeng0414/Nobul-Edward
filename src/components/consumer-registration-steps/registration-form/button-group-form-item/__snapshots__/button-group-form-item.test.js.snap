// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroupFormItem Component Component Snapshot should match stored snapshot 1`] = `
<div
  className="consumer-registration-button-group"
>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <button
            className="ant-btn consumer-registration-button-item button-next ant-btn-primary ant-btn-lg"
            disabled={true}
            onClick={[Function]}
            type="submit"
          >
            <span>
              Submitted
            </span>
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;
